### PR TITLE
Add a software rasterizer backend

### PR DESCRIPTION
**Pull request by abarth**
We'll use this when bringing up Flutter on Fuchsia.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: sky/shell/gpu/software/rasterizer_software.cc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  // There is no way for the compositor to know how long the layer tree
+  // construction took. Fortunately, the layer tree does. Grab that time
+  // for instrumentation.
+  compositor_context_.engine_time().SetLapTime(layer_tree->construction_time());
+
+  {
+    SkImageInfo info = SkImageInfo::MakeN32Premul(size.width(), size.height());
+    sk_sp<SkBitmapDevice> device(SkBitmapDevice::Create(info));
+    sk_sp<SkCanvas> canvas = sk_make_sp<SkCanvas>(device.get());
```
Comment (chinmaygarde) [inline sky/shell/gpu/software/rasterizer_software.cc]: We don't do anything with this canvas :(

File: sky/shell/gpu/software/rasterizer_software.cc [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  platform_view_ = nullptr;
+  teardown_completion_event->Signal();
+}
+
+// sky::shell::Rasterizer override.
+flow::LayerTree* RasterizerSoftware::GetLastLayerTree() {
+  return last_layer_tree_.get();
+}
+
+void RasterizerSoftware::Draw(uint64_t layer_tree_ptr,
```
Comment (chinmaygarde) [inline sky/shell/gpu/software/rasterizer_software.cc]: A lot of this code is completely duplicated in `rasterizer_direct`. Maybe we should move it into a common base class (`rasterizer_local` or something) and only have compositor access be detailed here. Just a thought. And maybe in a later patch.

File: sky/shell/gpu/software/rasterizer_software.cc (hunk 1 above)
Comment (abarth) [inline sky/shell/gpu/software/rasterizer_software.cc]: Yeah, we don't have a place to put the pixels after drawing them yet.  The fuchsia people are working on that part.  :)

File: sky/shell/BUILD.gn [hunk 3]
```diff
@@ -97,6 +97,8 @@ source_set("gpu_mojo") {
     "gpu/mojo/gl_texture_recycler.h",
     "gpu/mojo/rasterizer_mojo.cc",
     "gpu/mojo/rasterizer_mojo.h",
+    "gpu/software/rasterizer_software.cc",
```
Comment (chinmaygarde) [inline sky/shell/BUILD.gn]: Is `gpu_mojo` the right target for the software rasterizer?

**Comment by chinmaygarde** [comment]
lgtm.

File: sky/shell/gpu/software/rasterizer_software.cc (hunk 2 above)
Comment (abarth) [inline sky/shell/gpu/software/rasterizer_software.cc]: Yeah, I'm not sure how much that will stay that way.  Maybe it's too early to land this patch with so many unknowns.

**Pull request closed without merging**